### PR TITLE
feat: :sparkles: Adds storybook

### PR DESCRIPTION
**Pull request by AdrianValdes**
Installs Storybook initialized with Typescript, create-react-app was also installed using a Typescript template

### CODE REVIEW

**Comment by AdrianValdes** [comment]
Hi. I added the requested library. I am going to explain a little bit of the workflow I follow in case I am missing something.

1. On the main branch I did `git pull` to be sure I am up to date.
2. Create a new branch and checkout to the branch.
3. I installed the project.
4. I added unnecessary files to the `.gitignore`, cleaned the Readme.md, and deleted the `eject` from the `package.json` scripts.
5. I added the new storybook folder to git.
6. I commit the changes with a declarative message.
7. I went to Github an checked that one can merge the new branch with main without issues.
8. I checked I didn't commit unnecessary files.

Let me know, if I should change something please. Thanks a lot!

File: packages/storybook/package.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "storybook",
```
Comment (paschalidi) [inline packages/storybook/package.json:2]: In addition to the final name we give to this package I would also like to add a prefix. In the end the name would look like `@small-advertisments/<name-of-ui>`

File: packages/storybook/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# An UI-library created using Storybook
```
Comment (paschalidi) [inline packages/storybook/README.md:1]: good thinking here! 👍

File: packages/storybook/src/stories/Test.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+<Preview>
+  <Story name="button">
+    <Button>hello</Button>
+  </Story>
+</Preview>
+
+# Props
+
+<Props of={Button} />
```
Comment (paschalidi) [inline packages/storybook/src/stories/Test.stories.mdx:18]: This looks strange, I would imagine that stories would usually be a `tsx` type of file. 
Was it autogenerated or did we add it ouselves?

**Review by paschalidi — CHANGES_REQUESTED** [review]
Hey, good progrss you have done here. Most of the things look alright.  There are few things I have noticed and you might wanna have a look into them. 
* You might also want to delete the `.keepme` files since they dont serve any need at the moment.
* Also in the packages you have correctly added the `storybook` folder. In the package called `storybook` we will develop our user interface components. Now the name `storybook` for me doesnt really represent the fact that this package is for our user interface components. Understand that storybook is a library that we are using. We are not developing storybook we are developing some ui components. What I am trying to say is that the name should be more representative of _what_ we actually develop and not which libraries we are using to do so.
* One last thing that is related to the comment I made about deleting the `yarn start` and `yarn build`  scripts. I think it might be more clear after reading that comment that we dont want to be running the this package as a react app using the `react-scripts` but we want to run this app only using the storybook script. Now that means we wont be needing the `src/App.tsx`, `App.test.tsx`, `reportWebVitals.ts` and `src/index.tsx`.

Keep the good work up and poke me if there are questions 😄

**Comment by paschalidi** [comment]
Forgot to say, the steps you describe sound legit :)

File: packages/storybook/package.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    "@types/react-dom": "^16.9.10",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-scripts": "4.0.1",
+    "typescript": "^4.1.3",
+    "web-vitals": "^0.2.4"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
```
[the remark this answers is not in the document]
Comment (AdrianValdes) [reply packages/storybook/package.json:22]: I am understanding more and more little by little. The structure of the project is clearer to me in each step. Thanks for having the patience. 

I thought we would run the react-scripts to check how our components would look fast, like a playground. But I suppose that with Storybook and our own app it is enough.

File: packages/storybook/package.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    "test": "react-scripts test",
+    "storybook": "start-storybook -p 6006 -s public",
+    "build-storybook": "build-storybook -s public"
```
[the remark this answers is not in the document]
Comment (AdrianValdes) [reply packages/storybook/package.json:25]: Makes a lot of sense, I got it.

File: packages/storybook/package.json (hunk 1 above)
Comment (AdrianValdes) [reply packages/storybook/package.json:2]: Yes, this part makes sense too. I changed the name already. I couldn't think of a better name, we can change it later or if you have a better suggestion.

File: packages/storybook/src/stories/Test.stories.mdx (hunk 3 above)
Comment (AdrianValdes) [reply packages/storybook/src/stories/Test.stories.mdx:18]: That was a "Test" I added to check the `.mdx` files were being displayed correctly.

File: packages/simple-web-components/package.json [hunk 6]
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "@small-advertisements/simple-web-components",
```
Comment (paschalidi) [inline packages/simple-web-components/package.json:2]:
```suggestion
  "name": "@small-advertisements/ui",
```

Comment (AdrianValdes) [reply packages/simple-web-components/package.json]: Done

Comment (paschalidi) [reply packages/simple-web-components/package.json]: Can you also rename the whole file once again.

Comment (paschalidi) [reply packages/simple-web-components/package.json]: I mean the file in the `packages` to match the name we gave here, aka `ui`

**Comment by paschalidi** [comment]
Good question. Well we will do most if not all of the things in this article. 
We will have a components file and each component will have its own story and most of the things that are mentioned there we will do. We will do snapshot testing in addition to these practises. :) 

Does this answer your question?

File: packages/simple-web-components/package.json (hunk 6 above)
Comment (AdrianValdes) [reply packages/simple-web-components/package.json]: I thought about it but I wasn't 100 % sure. Done

Comment (paschalidi) [reply packages/simple-web-components/package.json]: @AdrianValdes check this ☝️

**Comment by AdrianValdes** [comment]
It does answer it. I don't what snapshot testing is, but I am already looking it up.

**Comment by paschalidi** [comment]
Basically in the future these three files will be in the same folder called `Button`. 

image

But for now all we wanted is storybook to run. Which you very well accomplished already ;)

**Review by paschalidi — APPROVED** [review]
This is good to go. 👍 Merge this in

**Comment by AdrianValdes** [comment]
I see, it makes sense. I am reading about how to integrate `styled-components` now. That the project is in Typescript makes all those steps a little longer.

**Comment by paschalidi** [comment]
Sorry last thing. Seems you the `src/.keepme` is still with us. Can you have a look please?

**Comment by AdrianValdes** [comment]
Done

**Comment by paschalidi** [comment]
No need to rush @AdrianValdes. We shape things well while we are working on them. And this is what is important for now. Prefer doing things well than doing them fast and not so great ;)

**Pull request merged**